traitlets/traitlets.py

```diff
@@ -1439,7 +1439,24 @@ def has_trait(self, name):
         return isinstance(getattr(self.__class__, name, None), TraitType)
 
     def has_trait_value(self, name):
-        """Return True if the specified trait has a value (not it's default)"""
+        """Return sTrue if the specified trait has a value.
+
+        This will return false even if ``getattr`` would return a
+        dynamically generated default value. These default values
+        will be recognized as existing only after they have been
+        generated.
+
+        Example
+
+        .. code-block:: python
```

Comment:
Since there's a high cost to every new public API method we are inserting on everybody's objects, maybe try to start every new trait method with `trait_`?